### ISSUE TITLE
[Kafka Microservice] UseInterceptors brake consistent processing of Kafka messages

### ISSUE THREAD

**Issue by slyder** [nestjs/nest#6254]
## Current behavior
Adding UseInterceptors to MessagePattern handler brakes consistent processing of Kafka messages. And consumer starts getting new messages without awaiting until completed currently run.

## Input Code

**main.ts**
```ts
async function bootstrap() {
  const app = await NestFactory.createMicroservice<MicroserviceOptions>(AppModule, {
    transport: Transport.KAFKA,
    options: {
      client: {
        brokers: ['localhost:9092'],
        clientId: 'social',
      },
      consumer: {
        groupId: 'social-group',
      }
    }
  });

  app.listen(() => console.log('Kafka consumer service is listening!'))
}
```

**app.controller.ts**
```ts
@Controller()
export class AppController {
  constructor(private readonly appService: AppService) { }

  @MessagePattern('ava.facebook.openBrowser')
  @UseInterceptors(LoggingInterceptor)
  async getHello(@Payload() message) {
    console.log('>>>>>>')
    await new Promise(resolve => setTimeout(resolve, 3000));
    console.log('<<<<')
    return 'Hello World';
  }
}
```

**logging.interceptor.ts**
```ts
@Injectable()
export class LoggingInterceptor implements NestInterceptor {
  intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
    // console.log('Before...');
    return next.handle()
  }
}
```

## Expected behavior
Output should be consistent:
```
>>>>>>
<<<<
>>>>>>
<<<<
>>>>>>
<<<<
```
But  after adding   _@UseInterceptors(LoggingInterceptor)_  you will receive invalid processing - consumer starts receive messages one right one, without awaiting for completion of previous task
```
>>>>>>
>>>>>>
>>>>>>
<<<<
<<<<
<<<<
```

**Comment by kamilmysliwiec** [comment]
Please provide a minimum reproduction repository (in the best-case scenario, with the `docker-compose` file which reproduces your Kafka setup).

**Comment by slyder** [comment]
@kamilmysliwiec https://github.com/slyder/nodejs-kafka-consumer-bug - commited code to GH

_Steps for testing:_

**docker shell**
docker-compose up -d

docker-compose exec kafka bash
/opt/kafka/bin/kafka-console-producer.sh --bootstrap-server 127.0.0.1:9092 \
  --topic test-topic < /producer-data.txt

**application shell**
nest start



In repo i have commented:
// @UseInterceptors(LoggingInterceptor)
so after producing messages in docker shell you will receive concurrent output:
````
Kafka consumer service is listening!
>>>>>>
<<<<
>>>>>>
<<<<
```

But if you uncomment UseInterceptors and restart application messages will be processed in invalid order:
````
>>>>>>
>>>>>>
<<<<
<<<<
````

**Comment by kamilmysliwiec** [comment]
@mkaufmaner would you like to take a look at this issue?

**Comment by slyder** [comment]
Hello! Is there any updates? =)

**Comment by vinimdocarmo** [comment]
I am having the same problem. I've spent some time trying to see what was going on and just now I was able to find this issue here.

**Comment by alexandr2110pro** [comment]
@vinimdocarmo is that a secret? :)

**Comment by vinimdocarmo** [comment]
@alexandr2110pro what do you mean?

**Comment by mustafa-barakat** [comment]
I replicated the bug but not using your code
used default interceptor sample

```

import { Injectable, NestInterceptor, ExecutionContext, CallHandler } from '@nestjs/common';
import { Observable } from 'rxjs';
import { tap } from 'rxjs/operators';

@Injectable()
export class LoggingInterceptor implements NestInterceptor {
  intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
    console.log('Before...');

    const now = Date.now();
    return next
      .handle()
      .pipe(
        tap(() => console.log(`After... ${Date.now() - now}ms`)),
      );
  }
}

```
and Added the interceptor to the controller function
`  @UseInterceptors(LoggingInterceptor)
`
got the same issue
```

Before...
>>>>>>
Before...
>>>>>>
Before...
>>>>>>
<<<<
After... 3025ms
<<<<
After... 3016ms
<<<<
After... 3016ms
```

**Comment by mustafa-barakat** [comment]
Also i checked Kafka partitions, all messages arrive to same parition, so no logical reason why this is happening

**Comment by lazaroofarrill** [comment]
I don't understand why would you expect the messages to execute in a queue in an async function. I would not expect that incoming requests await ongoing ones, that would break parallelism. The question should be why this is not happening without the interceptor.

**Comment by mustafa-barakat** [comment]
> I don't understand why would you expect the messages to execute in a queue in an async function. I would not expect that incoming requests await ongoing ones, that would break parallelism. The question should be why this is not happening without the interceptor.

It should as they are all the same message pattern ? if so, you're right, sth is wrong without the interceptor not with it

**Comment by sheepster1** [comment]
Facing a similar issue, any updates?

**Comment by kamilmysliwiec** [comment]
This has been changed in this PR https://github.com/nestjs/nest/pull/9586 and will be published as part of v9.0.o
